perfeed/git_providers/github.py

```diff
@@ -210,7 +210,99 @@ async def list_pr_numbers(
+
+        Args:
+            repo_name (str): The name of the repository.
+            start_date (datetime): The start date for filtering PRs.
+            end_date (datetime): The end date for filtering PRs.
+
+        Returns:
+            list[dict]: A list of pull request dictionaries.
+        """
+        all_prs = []
+        page = 1
+
+        while True:
+            # Fetch a page of pull requests (disable parallelism and handle pagination manually)
+            prs = self.api.pulls.list(owner=self.owner, repo=repo_name, state="closed", sort="created", direction="desc", per_page=100, page=page)
+            
+            # If no PRs were returned, break out of the loop
+            if not prs:
+                break
 
+            # Filter PRs for the date range within this page
+            filtered_prs = [
+                pr for pr in prs
+                if start_date <= datetime.strptime(pr['created_at'], '%Y-%m-%dT%H:%M:%SZ') <= end_date
+            ]
+            
+            all_prs.extend(filtered_prs)
+
+            # If fewer than 100 PRs were returned, this is the last page
+            if len(prs) < 100:
+                break
+
+            # Move to the next page
+            page += 1
+
+        return all_prs
+
+    async def get_authors_repo_prs(self, authors: list[str], start_date: datetime, end_date: datetime) -> list[AuthorRepositoryPullRequests]:
+        """
+        Fetch pull requests grouped by author and repository.
+        
+        Args:
+            authors (list[str]): List of author usernames to fetch PRs for.
+            start_date (datetime): The start date for filtering PRs.
+            end_date (datetime): The end date for filtering PRs.
+
+        Returns:
+            list[AuthorRepositoryPullRequests]: A list of dataclass objects containing PR data grouped by author.
+        """
+        author_prs_map = {}
+        
+        # Manual pagination to fetch repositories without parallelism
+        all_repos = []
+        page = 1
+        while True:
+            repos = self.api.repos.list_for_org(org=self.owner, per_page=100, page=page)
+            if not repos:
+                break
+            all_repos.extend(repos)
+            if len(repos) < 100:
+                break
+            page += 1
+        
+        # Fetch PRs for each repository and group by author
+        for repo in all_repos:
+            prs = await self._fetch_pr_numbers(repo_name=repo.name, start_date=start_date, end_date=end_date)
+            for author in authors:
+                for pr in prs:
+                    if pr['user']['login'] == author:
+                        if author not in author_prs_map:
+                            author_prs_map[author] = {}
+                        if repo.name not in author_prs_map[author]:
+                            author_prs_map[author][repo.name] = []
+                        author_prs_map[author][repo.name].append(pr['number'])
```

Comment:
got a feeling you can reduce this layers using defaultdict from python
`
from collections import defaultdict
author_prs_map = defaultdict(lambda: defaultdict(list))
`
replace the lines with
`author_prs_map[author][repo.name].append(pr['number'])`